src/Chess/ChessBoard.hs

```diff
@@ -29,9 +29,27 @@ chessBoardRowAsString rowNum chessBoard = rowNumStr ++ "  " ++ (foldl (\ rowStr
 -- If there is a piece at the destination square, then that piece will be replaced
 -- If the move results in a check-mate, sets the state of the ChessBoard to Over
 -- WARNING: Assumes given move is valid!
--- TODO: implement + test this (1.5 hours) [Yiyi]
+-- TODO: test this [Yiyi]
 makeMove :: ChessBoard -> ChessMove -> ChessBoard
-makeMove chessBoard move = chessBoard
+makeMove cb@(ChessBoard pieces state) (ChessMove from to) = 
+    ChessBoard (putPiece pieceToMove to (removePiece from pieces)) (if (isCheckMate cb (getPieceColour pieceToMove)) then Over else state)
```

Comment:
I think you're still missing the second `removePiece`. See my previous comment.

Reply:
i checked it in the `putPiece` instead since we are passing the `to`, wouldn’t it work?

Reply:
ah I missed that! Yep, that should work :)